### PR TITLE
fix: get:url pin corrupts container_client on other tagged instances

### PR DESCRIPTION
**Pull request by Rarfael**
## Summary

Loading two tagged instances of this plugin (one per Azure Blob container, the documented way to use more than one container) and calling `get:url` for one container silently corrupts a different tagged instance's subsequent `save$`/`load$`, pointing it at the wrong container. No error at any point: every individual Azure call succeeds, just against the wrong container.

## Root cause

`getSignedUrl` (the handler for `get:url`) calls `container_check(container)`, which reassigns the `container_client` variable that `save`/`load`/`remove` also use. But `getSignedUrl` never actually uses `container_client` after that call; it builds its own separate client on the next line. So the call does nothing for `get:url` itself, and only has the side effect of leaving `container_client` pointed at the wrong container for later saves and loads.

## Fix

Removes the dead `container_check` call and the now-unused `container_check` function. `getSignedUrl`'s own behavior is unchanged.

## Test

Adds a regression test: load two tagged instances, mint a SAS for container A, save an entity that belongs to instance B, and check it landed in B, not A. This test fails on the code before the fix and passes after.

### CODE REVIEW

File: test/blob-store.test.js [hunk 1]
```diff
@@ -341,6 +341,73 @@ lab.test('bin-local-customid', async function () {
...
+  // Save an entity that belongs to instance B. Before the fix, this could
+  // land in container A instead of container B.
+  const saved = await s0
+    .entity('regress/b')
+    .data$({ content: Buffer.from('instance-b-content') })
+    .save$({ id$: 'regress-test.bin' })
+
+  expect(saved.entity$).equal('-/regress/b')
+
+  const loadedB = await s0.entity('regress/b').load$('regress-test.bin')
```
Comment (BeAllAround) [inline test/blob-store.test.js:402]: Very good catch! The `make_blob_id` function is not unique to the `container_name` - the container name assigned at the point of plugin initialization.

Comment (BeAllAround) [reply test/blob-store.test.js:402]: This means `regress/b` will actually be saved to `container_client` of `test-container-regress-a` after `cloud:azure,service:store,get:url,kind:upload` overrides the `container_client` closure variable.

Because the `make_blob_id` function does not make the id based on the `canon` + `id `+ `container`.

Both of these are then saved to `regress-test.bin`, which just happened to map into `regress/b` and `regress/a` due to the `replyEnt` function. So the explicit id of  `regress-test.bin` will get it to the same point of memory/storage.

File: src/blob-store.ts [hunk 2]
```diff
@@ -371,8 +367,6 @@ async function blob_store(this: any, options: any) {
       let accessUrl = '',
         sasToken = ''
 
-      await container_check(container)
-
```
Comment (BeAllAround) [inline src/blob-store.ts:375]: The whole bug resolves around the JavaScript closure where the `getSignedUrl` is registered by `seneca.message` on the next load of the plugin instance. So, essentially if `#regress_a`, `#regress_b`, and `#regress_c` are loaded, the scope of the `getSignedUrl` looks up the `container_client` of `#regress_c`  - which was loaded the last after `seneca.ready`.

This means that given the comments below, the fix is to remove the `container_check` function and but we still are required to check if the container exists so, please update the PR with the following code:

Instead of

`const containerClient = blob_client.getContainerClient(container)`

Update it with:

`const containerClient = await load_container_client(container)`.

This isolates the closure variable where we look the `containerClient` from the current scope now.

This should have your edge case test pass now while we are still checking if the container exists!

That'd be all and let's merge this!

Great job finding this!

Comment (Rarfael) [reply src/blob-store.ts:375]: Thank you. Already added the change you asked for
